From 51d4c25b72f97b534951d15685b671a512bf5cef Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 7 Mar 2017 16:26:30 +0000
Subject: [PATCH 0827/1077] MANTIS 45083 Mantis 45083: ConveyorDev.idl with
 changed SubDevice status The subdevice moduleStatus.status is now of
 errorType

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1164 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl         | 19 ++++++-------------
 1 file changed, 6 insertions(+), 13 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index 0b2b285a..6cfaf114 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,7 +39,7 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.13";
+   const string interface_version = "1.14";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
@@ -185,28 +185,21 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       ERR_BACK_INTRUSION,           /**< Back-side safety intrusion. */
       ERR_PIECE_MISTRACKED,         /**< Baggage tracking failure. */
       ERR_MANUAL_MODE,              /**< Online, manual mode */
-      ERR_SWITCHED_OFF              /**< Online, switched off */
+      ERR_SWITCHED_OFF,             /**< Online, switched off */
+      ERR_HARDWARE_DEGRADED         /**< Hardware partial broken */
    };
 
+
    /**
     *
     * Definition of status of a specific device in the baggage handling system.
     *
    **/
-   enum subDeviceStatus
-   {
-      SD_OK,                        /**< Subdevice ok */
-      SD_DISCONNECTED,              /**< Subdevice is disconnected */
-      SD_HARDWARE_ERROR,            /**< Subdevice hardware error */
-      SD_HARDWARE_DEGRADED          /**< Subdevice warning "degraded" */
-   };
-
-
    struct moduleStatus
    {
       moduleCharacteristic type;    /**< Identify the particular device */
-      subDeviceStatus status;       /**< General device status */
-      string description;           /**< Human readable details */
+      errorType         status;     /**< This device's status */
+      string            description; /**< Human readable details */
    };
 
    typedef sequence<moduleStatus> moduleStatusList;
-- 
2.41.0.windows.1

